rfcs/0000-centralized-configuration-management.md

```diff
@@ -0,0 +1,384 @@
+- terraform resources and data sources with the aws provider
+- ansible workflows through the `community.aws.aws_ssm_parameter_store` module
+
+### Parameter Formatting
+
+We take advantage of Parameter Store's hierarchical nature by formatting the hierarchies into 4-tuple prefixes, culminating in a named _leaf_ node. Together, the 4-tuple prefix and leaf make up a single parameter: `/${project}/${env}/${group}/${sensitivity}/${leaf}`.
+Each member of the tuple represents a scoped level or tier providing increasingly specific context for BFD services and supporting systems.
+The scopes lend themselves well to crafting simple queries and tailored access policies that target the necessary hierarchies of configuration data only.
+The primary use cases for these parameters include infrastructure-as-code, automated-configuration-management, and release-management strategies in `terraform`, `ansible`, and `jenkins` tooling, respectively.
+This proposed format **does not** apply to other potential Parameter Store use-cases.
+
+### Project Scope
+
+Project scope (`${project}`) is the _parameter name root_. In these hierarchies, this is *practically* an invariant that resolves to our project: `bfd`.
+Nevertheless, this is necessary to separate bfd-specific resources from other potential hierarchies managed by external teams supporting wider CMS projects.
+
+#### Conventions
+
+- No leaves are allowed within this scope, e.g. `/bfd/foo` is illegal
+- Must resolve to `bfd`
+
+### Environment Scope
+
+Environment scope (`${env}`) includes typical Software Development Life Cycle (SDLC) Environments like production (`prod`), those used along the *path-to-production* like `test` and `prod-sbx`, as well as higher order environments that fall outside of the typical BFD release process, i.e. `mgmt`.
+This fundamental scope separates production hierarchies from non-production ones and facilitates simple, least-privilege role-based access controls for developers and services.
+
+#### Conventions
+
+- No leaf nodes are allowed within this scope, e.g. `/bfd/prod/foo` is illegal
+- Must be one of `mgmt`, `prod`, `prod-sbx`, or `test`
+
+#### Examples
+
+- `/bfd/mgmt/`
+- `/bfd/prod/`
+- `/bfd/prod-sbx/`
+- `/bfd/test/`
+
+### Group Scope
+
+Group scope (`${group}`) largely provides the service-level configuration. The more inclusive *group* concept is used instead of *service* because hierarchies may include scopes that don't map to any one service.
+
+#### Conventions
+
+- No leaf nodes are allowed within this scope, e.g. `/bfd/test/common/foo` is illegal
+- Must be one of `jenkins`, `common`, `migrator`, `pipeline`, or `server`
+
+#### Examples
+
+- `/bfd/mgmt/jenkins/`
+- `/bfd/prod-sbx/migrator/`
+- `/bfd/prod/server/`
+- `/bfd/test/common/`
+
+### Sensitivity Scope and the Leaf Node
+
+Sensitivity scope (`${sensitivity}`) defines whether a leaf node (`${leaf}`) in the hierarchy is stored as an ordinary `String` or encrypted with AWS KMS as a `SecureString`.
```

Comment:
As far as KMS, will there be a single key or multiple keys predicated on the env scope?